process_pr.py

```diff
@@ -1464,88 +1471,198 @@ def process_pr(repo_config, gh, repo, issue, dryRun, cmsbuild_user=None, force=F
+                    }
                     if len(commit.parents) > 1:
                         bot_cache[commit.sha]["files"] = []
                     else:
                         bot_cache[commit.sha]["files"] = sorted(
                             x["filename"] for x in get_commit(repo.full_name, commit.sha)["files"]
                         )
+
                 elif len(commit.parents) > 1:
                     bot_cache[commit.sha]["files"] = []
 
-                if commit.sha not in bot_cache:
-                    bot_cache[commit.sha] = {
-                        "time": int(commit.commit.committer.date.timestamp()),
-                        "files": sorted(
-                            x["filename"] for x in get_commit(repo.full_name, commit.sha)["files"]
-                        ),
-                    }
-
                 cache_entry = bot_cache[commit.sha]
                 events[datetime.fromtimestamp(cache_entry["time"])].append(
-                    {
-                        "type": "commit",
-                        "value": cache_entry["files"],
-                    }
+                    {"type": "commit", "value": {"files": cache_entry["files"], "sha": commit.sha}}
                 )
 
+            # Inject events from cached commits
+            for commit_sha, cache_entry in commit_cache.items():
+                if cache_entry.get("squashed", False):
+                    print("Adding back cached commit {0}".format(commit_sha))
+                    events[datetime.fromtimestamp(cache_entry["time"])].append(
+                        {
+                            "type": "commit",
+                            "value": {"files": cache_entry["files"], "sha": commit_sha},
+                        }
+                    )
+
     # Process commits and signatures
+    signed_commit_sha = ""
+    flattened_eventlist = []
     for _, eventlist in sorted(events.items()):
-        for event in eventlist:
-            print("Event:", event)
-            if event["type"] == "sign":
-                selected_cats = event["value"]["selected_cats"]
-                ctype = event["value"]["ctype"]
-                if any(x in signing_categories for x in selected_cats):
-                    set_comment_emoji_cache(
-                        dryRun, bot_cache, event["value"]["comment"], repository
+        flattened_eventlist.extend(eventlist)
+
+    for event in flattened_eventlist:
+        print("Event:", event)
+        if event["type"] == "sign":
+            comment = event["value"]["comment"]
+            cached_signed_comments = bot_cache["signatures"].get(comment.id)
+            if cached_signed_comments and cached_signed_comments != signed_commit_sha:
+                print(
+                    "WARNING: For comment {0}, cached list of signed commits doesn't match the present list of signed commits. All signatures will be reset."
+                )
+                reset_all_signatures = True
+                break
```

Comment:
@iarspider , we should not stop processing the event loop here. We can set thumb's down (so some other) emoji for this comment and ignore this comment. We still want L2 to sign such PRs ... right? So if we break the loop then how can L2 sign it